### ISSUE TITLE
Auro-deflate injects Content-Encoding even wihtout body

### ISSUE THREAD

**Issue by ixti** [httprb/http#521]
Example:

```
require "http"
require "logger"

HTTP.use(:logging => { :logger => Logger.new(STDOUT) })
  .use(:auto_deflate)
  .get("https://example.com")
```

The above will send request:

```
GET https://example.com/
Connection: close
Content-Encoding: gzip
Host: example.com
User-Agent: http.rb/4.0.3
```

Content-Encoding is being sent as `gzip`, while there's no content at all. That makes some servers go nuts. E.g. I've seen 403 responses, while `example.com` responds with `400` (which I believe is correct response if it needs to be failed)

**Comment by ixti** [comment]
Another problem I encounter, which worth keeping in mind that some servers may respond with (!) 400 error when there's body and no content-length header, which is not present when auto-deflate is in use.

**Comment by Bonias** [comment]
> Content-Encoding is being sent as `gzip`, while there's no content at all

Fixed in #529 

---

> when there's body and no content-length header, which is not present when auto-deflate is in use

Can you provide an example? Content-Length is set when auto-deflate is in use:

```ruby
HTTP.use(logging: { logger: Logger.new(STDOUT) }).
  use(:auto_deflate).
  post("http://localhost:3000", body: 'xxx')
```

Here is what is logged by logger:

```
I, [2019-02-12T21:46:37.062172 #52955]  INFO -- : > POST http://localhost:3000/
D, [2019-02-12T21:46:37.062277 #52955] DEBUG -- : Connection: close
Content-Encoding: gzip
Host: localhost:3000
User-Agent: http.rb/5.0.0.pre

xxx
```

and here is what is actually sent:

```
POST / HTTP/1.1
Connection: close
Content-Encoding: gzip
Host: localhost:3000
User-Agent: http.rb/5.0.0.pre
Content-Length: 23

�0c\���
�
```

**Comment by ixti** [comment]
Thank for the fix. About Content-Length with body, indeed, I have overlooked that it's simply not being logged, so my problem was surely related to something else. Thanks again for looking into this.
